### PR TITLE
feat(iotda): add a datasource to get the list of device linkage rules

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a datasource to get the list of device linkage rules.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource to get the list of the device linkage rules.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Basic
```
$ export HW_REGION_NAME=cn-north-4 
$ unset HW_IOTDA_ACCESS_ADDRESS
$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDataSourceDeviceLinkageRules_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDataSourceDeviceLinkageRules_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDeviceLinkageRules_basic
=== PAUSE TestAccDataSourceDeviceLinkageRules_basic
=== CONT  TestAccDataSourceDeviceLinkageRules_basic
--- PASS: TestAccDataSourceDeviceLinkageRules_basic (122.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     122.828s
```
## Skip
```
$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDataSourceDeviceLinkageRules_derived"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDataSourceDeviceLinkageRules_derived -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDeviceLinkageRules_derived
=== PAUSE TestAccDataSourceDeviceLinkageRules_derived
=== CONT  TestAccDataSourceDeviceLinkageRules_derived
    acceptance.go:1414: HW_IOTDA_ACCESS_ADDRESS must be set for the acceptance test
--- SKIP: TestAccDataSourceDeviceLinkageRules_derived (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     0.041s
```
## Derived
```
$ export HW_REGION_NAME=cn-south-1
$ export HW_IOTDA_ACCESS_ADDRESS=e333xxxxxx.st1.iotda-app.cn-south-1.myhuaweicloud.com
$ make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccDataSourceDeviceLinkageRules_derived"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccDataSourceDeviceLinkageRules_derived -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDeviceLinkageRules_derived
=== PAUSE TestAccDataSourceDeviceLinkageRules_derived
=== CONT  TestAccDataSourceDeviceLinkageRules_derived
--- PASS: TestAccDataSourceDeviceLinkageRules_derived (100.91s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     100.953s
```

### CODE REVIEW

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+* `duration` - The duration of the device status, in minutes.
+  The valid value is range from `0` to `60`.
+
+<a name="rule_actions"></a>
+The `actions` block supports:
+
+* `type` - The rule action type of the device linkage rule. The valid values are as follows:
+  + **DEVICE_CMD**: Deliver commands.
+  + **SMN_FORWARDING**: Send SMN notifications.
+  + **DEVICE_ALARM**: Report alarms or clear alarms.
```
Comment (jinyangyang222) [inline docs/data-sources/iotda_device_linkage_rules.md:176]: Inconsistent with API documentation description?

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+* `project_id` - The project ID to which the SMN belongs.
+
+<a name="action_device_alarm"></a>
+The `device_alarm` block supports:
+
+* `name` - The name of the alarm.
+
+* `type` - The type of the alarm. The valid values are as follows:
+  + **fault**: Report alarms.
+  + **recovery**: Clear alarms.
```
Comment (jinyangyang222) [inline docs/data-sources/iotda_device_linkage_rules.md:220]: Restore alarms

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+* `severity` - The severity level of the alarm.
+  The valid values can be **warning**, **minor**, **major** or **critical**.
```
Comment (jinyangyang222) [inline docs/data-sources/iotda_device_linkage_rules.md:223]:
```suggestion
  The valid values can be **warning**, **minor**, **major**, or **critical**.
```

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+  The valid values are as follows:
+  + **DEVICE_LINKAGE**: Cloud based linkage rule.
+  + **DEVICE_SIDE**: Device side rule.
+
+* `status` - (Optional, String) Specifies the current status of the device linkage rule.
+  The valid values are as follows:
+  + **active**: The device linkage rule is active.
+  + **inactive**: The device linkage rule is not enabled.
+
+* `space_id` - (Optional, String) Specifies the ID of the resource space to which the device linkage rule belongs.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+* `name` - The name of the device linkage rule.
+
+* `description` - The description of the device linkage rule.
+
+* `type` - The type of the device linkage rules.
+
+* `status` - The current status of the device linkage rule.
+
+* `space_id` - The ID of the resource space to which the device linkage rules belong.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+* `device_id` - The ID of the device which trigger the rule.
+
+* `product_id` - The ID of the product associated with the device.
+
+* `path` - The path of the device property.
+  The format is **service_id/DataProperty**. e.g. **DoorWindow/status**
+
+* `operator` - The data comparison operator. The valid values are: **>**, **<**,
+  **>=**, **<=**, **=**, **in** and **between**.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+
+* `value` - The Rvalue of a data comparison expression.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+
+* `trigger_strategy` - The judgment strategy triggered by rule conditions. The valid values are:
+  + **pulse**: When the data reported by the device meets the conditions, the rule can be triggered.
+  + **reverse**: When the data reported by the device last time does not meet the conditions and the
+  data reported this time meets the conditions, the rule can be triggered.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+
+* `data_validatiy_period` - The data validity period, in seconds.
+
+<a name="trigger_simple_timer_condition"></a>
+The `simple_timer_condition` block supports:
+
+* `start_time` - The start time for triggering the rule.
+  The format is **yyyyMMdd'T'HHmmss'Z'**. e.g. **20151212T121212Z**.
+
+* `repeat_interval` - The interval for triggering the rule, in minutes.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+
+* `repeat_count` - The repeat times for triggering the rule.
+  The valid value is range from `1` to `9999`.
```
[the remark this answers is not in the document]
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md (hunk 1 above)
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md (hunk 2 above)
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

File: docs/data-sources/iotda_device_linkage_rules.md (hunk 3 above)
Comment (ruwenqiang123) [reply docs/data-sources/iotda_device_linkage_rules.md]: done

**Pull request merged**